### PR TITLE
osd: Making use of find to reduce computational complexity

### PR DESCRIPTION
**Pull request by shinobu-x**
Signed-off-by: Shinobu Kinjo <shinobu@redhat.com>

### CODE REVIEW

**Comment by shinobu-x** [comment]
test this please

File: src/osd/PrimaryLogPG.cc [hunk 1]
```diff
@@ -424,7 +424,7 @@ void PrimaryLogPG::on_local_recover(
       bool got = obc->get_recovery_read();
       assert(got);
 
-      assert(recovering.count(obc->obs.oi.soid));
+      assert(recovering.find(obc->obs.oi.soid) != recovering.end());
```
Comment (liewegas) [inline src/osd/PrimaryLogPG.cc]: This changes aren't helpful/useful.. the complexity is the same and they are more verbose.

The point is get rid of patterns like:

<pre>
if (foo.count(bar) && foo[bar] == baz) {
</pre>
where a count() precedes an operator[] call.  That is two lookups but can be reduced to 1 by using an iterator and comparing to end().

File: src/osd/PrimaryLogPG.cc [hunk 2]
```diff
@@ -755,7 +761,7 @@ void PrimaryLogPG::maybe_force_recovery()
        ++it) {
     if (*it == get_primary()) continue;
     pg_shard_t peer = *it;
-    if (peer_missing.count(peer) &&
+      if (peer_missing.find(peer) != peer_missing.end() &&
```
Comment (liewegas) [inline src/osd/PrimaryLogPG.cc]: Like here:
<pre>
auto it = peer_missing.find(peer);
if (it != peer_missing.end() &&
   !it->second.get_rmissing.empty()) {
...
</pre>

**Comment by shinobu-x** [comment]
@liewegas Please check when you get a chance.

**Comment by yuriw** [comment]
[wip-yuri2-testing-2018-01-05-1700](https://trello.com/c/BjEXSW3X/219-wip-yuri2-testing-2018-01-05-1700)

**Pull request merged**